packages/core/src/new-api/internal/execution/executionStateReducer.ts

```diff
@@ -0,0 +1,59 @@
+import { isDeploymentType } from "../../type-guards";
+import {
+  DeploymentExecutionState,
+  ExecutionState,
+  ExecutionStateMap,
+  ExecutionStatus,
+} from "../../types/execution-state";
+import { FutureStart, JournalableMessage } from "../../types/journal";
+
+export function executionStateReducer(
+  executionStateMap: ExecutionStateMap,
+  action: JournalableMessage
+) {
+  if (action.type === "execution-start") {
+    return {
+      ...executionStateMap,
+      [action.futureId]: initialiseExecutionStateFor(action),
+    };
+  }
+
+  if (action.type === "execution-success") {
+    const updatedExecutionState: DeploymentExecutionState = {
+      ...(executionStateMap[action.futureId] as DeploymentExecutionState),
+      status: ExecutionStatus.SUCCESS,
+      contractAddress: action.contractAddress,
+    };
+
+    return {
+      ...executionStateMap,
+      [action.futureId]: updatedExecutionState,
+    };
+  }
+
```

Comment:
Are the intermediate events between start and success missing on purpose?

Reply:
Yes, I want to tackle this while doing restartability: https://github.com/NomicFoundation/ignition/issues/258